app/services/issuing_ca_service.rb

```diff
@@ -0,0 +1,79 @@
+class IssuingCaService
+  CA_ISSUER_HOST_ALLOW_LIST = Figaro.env.ca_issuer_host_allow_list.split(',')
+
+  class UnexpectedPKCS7Response < StandardError; end
+
+  CA_RESPONSE_CACHE_EXPIRATION = 60.minutes
+
+  def self.fetch_signing_key_for_cert(cert)
+    return nil unless cert.aia && cert.aia['CA Issuers'].is_a?(Array)
+    ca_issuers = cert.aia['CA Issuers'].map do |issuer|
+      issuer = issuer.to_s
+      next unless issuer.starts_with?('URI')
+      issuer = issuer.gsub(/^URI:/, '')
+      uri = URI.parse(issuer)
+      next unless uri.scheme == 'http'
+      next unless allowed_host?(uri.host)
+      uri
+    end.compact
```

Comment:
a few things:
- Some ideas for avoiding `unless`
- Ideas for avoiding reassigning `issuer` param:
- possibly simpler control flow with fewer early returns?

this is subjective, just an idea

```suggestion
    ca_issuers = Array(cert.aia && cert.aia['CA Issuers']).map do |issuer|
      if issuer.to_s.starts_with?('URI:')
        uri = URI.parse(issuer.to_s.gsub(/^URI:/, ''))
        uri if uri.scheme == 'http' && allowed_host?(uri.host)
      end
    end.compact
```

Reply:
Acknowledging the subjectivity for myself as well on this. I'm not committed to what we have at the moment, but I have difficulty parsing nested calls and implicit returns while reading. It's lengthier, but the intention was a series of clear steps for filtering and transformation.

I was considering something like `filter_map`, but it would have to be `filter -> map -> filter` which felt harder to follow.

Reply:
there's `flat_map` which will remove `nil` values automatically?

Reply:
I think `flat_map` will only pull values out of arrays

```ruby
irb(main):003:0> ['a', nil, ['a']].flat_map { |x| x }
=> ["a", nil, "a"]
```

Reply:
Ah right, well probably fine to leave as-is for a bit then